### PR TITLE
Fix control data checking by capabilities

### PR DESCRIPTION
**Pull request by LitvinenkoIra**

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc [hunk 1]
```diff
@@ -193,9 +173,26 @@ bool CheckLightDataByCapabilities(
...
+  auto it = capabilities_status.asArray()->begin();
+  for (; it != capabilities_status.asArray()->end(); ++it) {
+    smart_objects::SmartObject& so = *it;
+    int64_t current_id = so[message_params::kName].asInt();
+    if (current_id == light_data[message_params::kId].asInt()) {
+      return CheckLightDataByCapabilities(so, light_data);
+    }
+  }
+  LOG4CXX_DEBUG(logger_, "There is no such light name in capabilities");
+  return false;
```
Comment (BSolonenko) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @LitvinenkoIra  I think, you can write : 
```
  auto begin = capabilities_status.asArray()->begin();
  auto end = capabilities_status.asArray()->end();
  auto it = std::find(begin, end, name);
  return it == end ? false : CheckLightDataByCapabilities(*it, light_data);
```

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc [hunk 2]
```diff
@@ -193,9 +173,26 @@ bool CheckLightDataByCapabilities(
   return true;
 }
 
+bool CheckLightNameByCapabilities(
+    const smart_objects::SmartObject& capabilities_status,
+    const smart_objects::SmartObject& light_data) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  auto it = capabilities_status.asArray()->begin();
+  for (; it != capabilities_status.asArray()->end(); ++it) {
+    smart_objects::SmartObject& so = *it;
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @LitvinenkoIra const?

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc [hunk 3]
```diff
@@ -193,9 +173,26 @@ bool CheckLightDataByCapabilities(
...
+    int64_t current_id = so[message_params::kName].asInt();
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @LitvinenkoIra const here is mandatory due to further comparison

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc [hunk 4]
```diff
@@ -193,9 +173,26 @@ bool CheckLightDataByCapabilities(
...
+    if (current_id == light_data[message_params::kId].asInt()) {
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @LitvinenkoIra swap comparable items

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc (hunk 1 above)
Comment (AByzhynar) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @BSolonenko Good point  but with this approach we have to sacrifice logging

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc [hunk 5]
```diff
@@ -231,6 +228,39 @@ bool CheckControlDataByCapabilities(
   return true;
 }
 
+bool CheckHmiControlDataByCapabilities(
+    const smart_objects::SmartObject& module_caps,
+    const smart_objects::SmartObject& control_data) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::map<std::string, std::string> mapping =
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @LitvinenkoIra const?

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc (hunk 1 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc]: @BSolonenko in this case capabilities_status is array of smart objects like 
```
[
    {
         "densityAvailable":true,
          "name":"FRONT_LEFT_HIGH_BEAM",
          "sRGBColorSpaceAvailable":true
    },
  ....
]
```
and i need to find appropriate name, so i think std::find will not work in this case

**Comment by LitvinenkoIra** [comment]
@BSolonenko @AByzhynar please look at d858a9a897092d6781cd3317d01dab51fae8f2db

**Pull request merged**